mlflow/recipes/steps/evaluate.py

```diff
@@ -136,107 +136,110 @@ def my_warn(*args, **kwargs):
             filename = frame.f_code.co_filename
             lineno = frame.f_lineno
             message = f"{timestamp} {filename}:{lineno}: {args[0]}\n"
-            open(os.path.join(output_directory, "warning_logs.txt"), "a").write(message)
+            with open(os.path.join(output_directory, "warning_logs.txt"), "a") as f:
+                f.write(message)
 
         original_warn = warnings.warn
         warnings.warn = my_warn
         try:
             import pandas as pd
 
-            open(os.path.join(output_directory, "warning_logs.txt"), "w")
+            with open(os.path.join(output_directory, "warning_logs.txt"), "w"):
+                self._validate_validation_criteria()
```

Comment:
```suggestion
            with open(os.path.join(output_directory, "warning_logs.txt"), "w"):
                pass

            self._validate_validation_criteria()
```

Reply:
Oh, I thought we wanted to keep opening the `warning_logs.txt`. Is it just for creating a log file? Then, let me change here.